src/coprocessor/dag/executor/table_scan.rs

```diff
@@ -74,17 +76,19 @@ impl TableScanExecutor {
             };
             let row_data = box_try!(table::cut_row(value, &self.col_ids));
             let h = box_try!(table::decode_handle(&key));
+            self.last_key = Some(key);
             return Ok(Some(Row::new(h, row_data)));
         }
         Ok(None)
     }
 
-    fn get_row_from_point(&mut self, range: KeyRange) -> Result<Option<Row>> {
-        let key = range.get_start();
-        let value = self.store.get(&Key::from_raw(key), &mut self.statistics)?;
+    fn get_row_from_point(&mut self, mut range: KeyRange) -> Result<Option<Row>> {
+        let key = range.take_start();
+        let value = self.store.get(&Key::from_raw(&key), &mut self.statistics)?;
         if let Some(value) = value {
             let values = box_try!(table::cut_row(value, &self.col_ids));
-            let h = box_try!(table::decode_handle(key));
+            let h = box_try!(table::decode_handle(&key));
+            self.last_key = Some(key);
```

Comment:
Should we set the last_key to None when we meet None?

Reply:
We don't need.